src/main/java/org/hawkular/client/android/util/Preferences.java

```diff
@@ -40,7 +41,14 @@ private Keys() {
         public static final String BACKEND_ENVIRONMENT = "environment";
     }
 
-    private SharedPreferences serverPreferences;
+    public static final class Defaults {
+        private Defaults() {
+        }
+
+        public static final int BACKEND_PORT = Integer.MIN_VALUE;
```

Comment:
Take a look at the `AuthorizationActivity#isPortCorrectNumber`. It is the invalid number used for checking purposes.